API.md

```diff
@@ -0,0 +1,518 @@
+# Sara Alert API
+
+Sara Alert provides a RESTful API to interact with the system to perform various actions. Actions include reading, writing, and updating monitoree data as well as reading monitoree lab results and monitoree daily reports. The data accepted and returned by the API corresponds to FHIR R4; the FHIR Patient resource is used for monitorees, the Observation FHIR resource is used for monitoree lab results, and the FHIR QuestionaireResponse FHIR resource is used for monitoree daily reports.
+
+- [CapabilityStatement](#cap)
+  - [GET [base]/fhir/r4/CapabilityStatement](#cap-get)
+- [Authenticating](#auth)
+  - [POST [base]/oauth/token](#auth-post)
+- [Reading](#read)
+  - [GET [base]/Patient/[:id]](#read-get-pat)
+  - [GET [base]/Observation/[:id]](#read-get-obs)
+  - [GET [base]/QuestionaireResponse/[:id]](#read-get-que)
+  - [GET [base]/Patient/[:id]/$everything](#read-get-all)
+- [Creating](#create)
+  - [POST [base]/Patient](#create-post-pat)
+- [Updating](#update)
+  - [PUT [base]/Patient/[:id]](#update-put-pat)
```

Comment:
Submitting an HTTP `DELETE [base]/fhir/r4/Patient/[id]` resulted in an "Action Controller: Exception caught" response containing a stack trace. In fact, getting stack traces exposed was not difficult. Exceptions should be caught and in those cases responses should be `OperationOutcome` resources.

Reply:
In this case, was it an `ActionController::RoutingError (No route matches [DELETE] ...`? In this case you would see the stack trace because you're in development mode. Regardless, exceptions should be handled in each route with an OperationOutcome. I've added this, but there is potential to improve the information returned in the response.